py/objringio.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Jim Mussared
```

Comment:
Maybe this should be your copyright instead, if you authored this entire file yourself?